Web Inspector: Worker: support debugging `requestAnimationFrame`

dcrousso opened the pull request:
#### 8a10ca4b355df596fe94964ccd7e443d7f3180df
<pre>
Web Inspector: Worker: support debugging `requestAnimationFrame`
<a href="https://bugs.webkit.org/show_bug.cgi?id=286667">https://bugs.webkit.org/show_bug.cgi?id=286667</a>

Reviewed by BJ Burg.

It&apos;s now possible to call `requestAnimationFrame` from inside a `Worker`, so Web Inspector should be able to set breakpoints for and capture async stack traces from them.

* Source/WebCore/workers/WorkerAnimationController.cpp:
(WebCore::WorkerAnimationController::requestAnimationFrame):
(WebCore::WorkerAnimationController::cancelAnimationFrame):
(WebCore::WorkerAnimationController::serviceRequestAnimationFrameCallbacks):
Add `InspectorInstrumentation` hooks for when
- `requestAnimationFrame` is called
- `cancelAnimationFrame` is called
- an animation frame callback is about to be invoked
- an animation frame callback just finished invoking

* Source/WebCore/inspector/InspectorInstrumentation.h:
(WebCore::InspectorInstrumentation::didRequestAnimationFrame):
(WebCore::InspectorInstrumentation::didCancelAnimationFrame):
(WebCore::InspectorInstrumentation::willFireAnimationFrame):
(WebCore::InspectorInstrumentation::didFireAnimationFrame):
* Source/WebCore/inspector/InspectorInstrumentation.cpp:
(WebCore::InspectorInstrumentation::didRequestAnimationFrameImpl):
(WebCore::InspectorInstrumentation::didCancelAnimationFrameImpl):
(WebCore::InspectorInstrumentation::willFireAnimationFrameImpl):
(WebCore::InspectorInstrumentation::didFireAnimationFrameImpl):
Support both `Document` and `DedicatedWorkerGlobalScope`.

* Source/WebCore/inspector/agents/InspectorDOMDebuggerAgent.h:
* Source/WebCore/inspector/agents/InspectorDOMDebuggerAgent.cpp:
(WebCore::InspectorDOMDebuggerAgent::disable):
(WebCore::InspectorDOMDebuggerAgent::mainFrameNavigated):
(WebCore::InspectorDOMDebuggerAgent::setEventBreakpoint):
(WebCore::InspectorDOMDebuggerAgent::removeEventBreakpoint):
(WebCore::InspectorDOMDebuggerAgent::willFireAnimationFrame):
(WebCore::InspectorDOMDebuggerAgent::didFireAnimationFrame):
* Source/WebCore/inspector/agents/page/PageDOMDebuggerAgent.h:
* Source/WebCore/inspector/agents/page/PageDOMDebuggerAgent.cpp:
(WebCore::PageDOMDebuggerAgent::disable):
(WebCore::PageDOMDebuggerAgent::mainFrameNavigated):
(WebCore::PageDOMDebuggerAgent::willFireAnimationFrame): Deleted.
(WebCore::PageDOMDebuggerAgent::didFireAnimationFrame): Deleted.
(WebCore::PageDOMDebuggerAgent::setAnimationFrameBreakpoint): Deleted.
* Source/WebCore/inspector/agents/worker/WorkerDOMDebuggerAgent.h:
* Source/WebCore/inspector/agents/worker/WorkerDOMDebuggerAgent.cpp:
(WebCore::WorkerDOMDebuggerAgent::setAnimationFrameBreakpoint): Deleted.
Move all `m_pauseOnAllAnimationFramesBreakpoint` code to the base class so it can be used inside `Worker`.

* Source/WebCore/inspector/agents/WebDebuggerAgent.h:
* Source/WebCore/inspector/agents/WebDebuggerAgent.cpp:
(WebCore::WebDebuggerAgent::didRequestAnimationFrame):
(WebCore::WebDebuggerAgent::willFireAnimationFrame):
(WebCore::WebDebuggerAgent::didCancelAnimationFrame):
(WebCore::WebDebuggerAgent::didFireAnimationFrame):
* Source/WebCore/inspector/agents/page/PageDebuggerAgent.h:
* Source/WebCore/inspector/agents/page/PageDebuggerAgent.cpp:
(WebCore::PageDebuggerAgent::didRequestAnimationFrame): Deleted.
(WebCore::PageDebuggerAgent::willFireAnimationFrame): Deleted.
(WebCore::PageDebuggerAgent::didCancelAnimationFrame): Deleted.
(WebCore::PageDebuggerAgent::didFireAnimationFrame): Deleted.
Move all animation frame async stack trace capture code to the base class so it can be used inside `Worker`.

* Source/WebInspectorUI/UserInterface/Controllers/DOMDebuggerManager.js:
(WI.DOMDebuggerManager.prototype._supportsAllAnimationFramesBreakpoint): Added.
(WI.DOMDebuggerManager.prototype._setEventBreakpoint):
(WI.DOMDebuggerManager.prototype._removeEventBreakpoint):
Allow the &quot;All Animation Frames&quot; breakpoint to be set on `Worker` targets.

* LayoutTests/inspector/worker/resources/worker-dom-debugger.js:
(functions.triggerAnimationFrame): Added.
* LayoutTests/inspector/worker/dom-debugger-event-animation-frame-breakpoints.html:
* LayoutTests/inspector/worker/dom-debugger-event-animation-frame-breakpoints-expected.txt:

* Source/JavaScriptCore/inspector/protocol/DOMDebugger.json:
* Source/WebInspectorUI/UserInterface/Protocol/Legacy/iOS/13.4/InspectorBackendCommands.js:
* Source/WebInspectorUI/UserInterface/Protocol/Legacy/iOS/14.0/InspectorBackendCommands.js:
* Source/WebInspectorUI/UserInterface/Protocol/Legacy/iOS/14.5/InspectorBackendCommands.js:
* Source/WebInspectorUI/UserInterface/Protocol/Legacy/iOS/15.0/InspectorBackendCommands.js:
* Source/WebInspectorUI/UserInterface/Protocol/Legacy/iOS/15.4/InspectorBackendCommands.js:
* Source/WebInspectorUI/UserInterface/Protocol/Legacy/iOS/16.0/InspectorBackendCommands.js:
* Source/WebInspectorUI/UserInterface/Protocol/Legacy/iOS/16.4/InspectorBackendCommands.js:
* Source/WebInspectorUI/UserInterface/Protocol/Legacy/iOS/17.0/InspectorBackendCommands.js:
* Source/WebInspectorUI/UserInterface/Protocol/Legacy/iOS/17.2/InspectorBackendCommands.js:
* Source/WebInspectorUI/UserInterface/Protocol/Legacy/iOS/17.4/InspectorBackendCommands.js:
* Source/WebInspectorUI/UserInterface/Protocol/Legacy/iOS/18.0/InspectorBackendCommands.js:
* Source/WebInspectorUI/UserInterface/Protocol/Legacy/iOS/18.2/InspectorBackendCommands.js:
* Source/WebInspectorUI/UserInterface/Protocol/Legacy/macOS/13.0/InspectorBackendCommands.js:
* Source/WebInspectorUI/UserInterface/Protocol/Legacy/macOS/13.3/InspectorBackendCommands.js:
* Source/WebInspectorUI/UserInterface/Protocol/Legacy/macOS/14.0/InspectorBackendCommands.js:
* Source/WebInspectorUI/UserInterface/Protocol/Legacy/macOS/14.2/InspectorBackendCommands.js:
* Source/WebInspectorUI/UserInterface/Protocol/Legacy/macOS/14.4/InspectorBackendCommands.js:
* Source/WebInspectorUI/UserInterface/Protocol/Legacy/macOS/15.0/InspectorBackendCommands.js:
* Source/WebInspectorUI/UserInterface/Protocol/Legacy/macOS/15.2/InspectorBackendCommands.js:
* Source/WebInspectorUI/Versions/Inspector-iOS-13.4.json:
* Source/WebInspectorUI/Versions/Inspector-iOS-14.0.json:
* Source/WebInspectorUI/Versions/Inspector-iOS-14.5.json:
* Source/WebInspectorUI/Versions/Inspector-iOS-15.0.json:
* Source/WebInspectorUI/Versions/Inspector-iOS-15.4.json:
* Source/WebInspectorUI/Versions/Inspector-iOS-16.0.json:
* Source/WebInspectorUI/Versions/Inspector-iOS-16.4.json:
* Source/WebInspectorUI/Versions/Inspector-iOS-17.0.json:
* Source/WebInspectorUI/Versions/Inspector-iOS-17.2.json:
* Source/WebInspectorUI/Versions/Inspector-iOS-17.4.json:
* Source/WebInspectorUI/Versions/Inspector-iOS-18.0.json:
* Source/WebInspectorUI/Versions/Inspector-iOS-18.2.json:
* Source/WebInspectorUI/Versions/Inspector-macOS-13.0.json:
* Source/WebInspectorUI/Versions/Inspector-macOS-13.3.json:
* Source/WebInspectorUI/Versions/Inspector-macOS-14.0.json:
* Source/WebInspectorUI/Versions/Inspector-macOS-14.2.json:
* Source/WebInspectorUI/Versions/Inspector-macOS-14.4.json:
* Source/WebInspectorUI/Versions/Inspector-macOS-15.0.json:
* Source/WebInspectorUI/Versions/Inspector-macOS-15.2.json:
* LayoutTests/inspector/worker/dom-debugger-dom-breakpoints.html: Removed.
* LayoutTests/inspector/worker/dom-debugger-dom-breakpoints-expected.txt: Removed.
* LayoutTests/platform/mac-site-isolation/TestExpectations:

Drive-by: mark `DOMDebugger.setDOMBreakpoint` and `DOMDebugger.removeDOMBreakpoint` as being only supported by `&quot;page&quot;` targets since that&apos;s how the code behaves (i.e. it&apos;ll throw an error if dispatched to a `Worker`).
Canonical link: <a href="https://commits.webkit.org/290245@main">https://commits.webkit.org/290245@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/5b37042560f957ebe0043dddcdeb6eb32dc1c488

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/89415 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/8940 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/44255 "Built successfully") | [  ~~🛠 wpe~~](https://ews-build.webkit.org/#/builders/5/builds/94402 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/40179 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/91466 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/9328 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/17218 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/5/builds/94402 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/59/builds/40179 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/92417 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/132/builds/7158 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/81146 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/5/builds/94402 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/133/builds/6899 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/64/builds/35529 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/39283 "Built successfully") | 
| [  ~~🛠 🧪 jsc~~](https://ews-build.webkit.org/#/builders/20/builds/82214 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/77256 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/36522 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/96230 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/88191 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/16596 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/12206 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/77756 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/16851 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/76941 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/77063 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/19014 "Built successfully and passed tests") | [  ~~🧪 vision-wk2~~](https://ews-build.webkit.org/#/builders/126/builds/21491 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-intel-wk2~~](https://ews-build.webkit.org/#/builders/119/builds/20078 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/9746 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/16609 "Built successfully") | [  ~~🛠 mac-safer-cpp~~](https://ews-build.webkit.org/#/builders/120/builds/21920 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/110684 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/16350 "Built successfully") | | [  ~~🧪 jsc-armv7-tests~~](https://ews-build.webkit.org/#/builders/25/builds/26539 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/129/builds/19801 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/18131 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

burg commented on LayoutTests/inspector/worker/dom-debugger-event-animation-frame-breakpoints.html:
```diff
@@ -11,30 +12,36 @@
     suite.addTestCase({
-        name: "Worker.DOMDebugger.EventBreakpoint.AnimationFrame.setEventBreakpoint",
-        description: "Check setting an event breakpoint on all animation frames is not supported for a Worker target.",
+        name: "Worker.DOMDebugger.EventBreakpoint.AnimationFrame.Basic",
+        description: "Check that adding/removing animation frame event breakpoints work for a Worker target.",
```
Wooo!

burg commented on Source/WebInspectorUI/UserInterface/Controllers/DOMDebuggerManager.js:
```diff
@@ -702,8 +702,10 @@ WI.DOMDebuggerManager = class DOMDebuggerManager extends WI.Object
 
-        // Worker targets do not support `requestAnimationFrame` breakpoints.
-        if (breakpoint === this._allAnimationFramesBreakpoint && target.type === WI.TargetType.Worker)
+        // COMPATIBILITY (iOS 18.X, macOS 15.X): Worker targets did not support `requestAnimationFrame` breakpoints yet.
```
Yuck but I guess we have this workaround at least. Can you pull the comment and repeated check into a helper function?

burg replied:
Also prefer .* instead of .X

dcrousso replied:
we've normally done `.X` in the past, and there's already an `18.X`/`15.X`, so id prefer to keep it as is at least for the next release so it's easier to search for when the time comes to change them.  totally fine switching to `.*` for any after the next release :)

alternatively we could use `IOS_TBA`/`MAC_TBA` to match what we do with `*_API_AVAILABLE` macros elsewhere in WebKit, but not sure if that'd confuse folks changing JSC/WK which we probably dont want

dcrousso committed:
Web Inspector: Worker: support debugging `requestAnimationFrame`
https://bugs.webkit.org/show_bug.cgi?id=286667

Reviewed by BJ Burg.

It's now possible to call `requestAnimationFrame` from inside a `Worker`, so Web Inspector should be able to set breakpoints for and capture async stack traces from them.

* Source/WebCore/workers/WorkerAnimationController.cpp:
(WebCore::WorkerAnimationController::requestAnimationFrame):
(WebCore::WorkerAnimationController::cancelAnimationFrame):
(WebCore::WorkerAnimationController::serviceRequestAnimationFrameCallbacks):
Add `InspectorInstrumentation` hooks for when
- `requestAnimationFrame` is called
- `cancelAnimationFrame` is called
- an animation frame callback is about to be invoked
- an animation frame callback just finished invoking

* Source/WebCore/inspector/InspectorInstrumentation.h:
(WebCore::InspectorInstrumentation::didRequestAnimationFrame):
(WebCore::InspectorInstrumentation::didCancelAnimationFrame):
(WebCore::InspectorInstrumentation::willFireAnimationFrame):
(WebCore::InspectorInstrumentation::didFireAnimationFrame):
* Source/WebCore/inspector/InspectorInstrumentation.cpp:
(WebCore::InspectorInstrumentation::didRequestAnimationFrameImpl):
(WebCore::InspectorInstrumentation::didCancelAnimationFrameImpl):
(WebCore::InspectorInstrumentation::willFireAnimationFrameImpl):
(WebCore::InspectorInstrumentation::didFireAnimationFrameImpl):
Support both `Document` and `DedicatedWorkerGlobalScope`.

* Source/WebCore/inspector/agents/InspectorDOMDebuggerAgent.h:
* Source/WebCore/inspector/agents/InspectorDOMDebuggerAgent.cpp:
(WebCore::InspectorDOMDebuggerAgent::disable):
(WebCore::InspectorDOMDebuggerAgent::mainFrameNavigated):
(WebCore::InspectorDOMDebuggerAgent::setEventBreakpoint):
(WebCore::InspectorDOMDebuggerAgent::removeEventBreakpoint):
(WebCore::InspectorDOMDebuggerAgent::willFireAnimationFrame):
(WebCore::InspectorDOMDebuggerAgent::didFireAnimationFrame):
* Source/WebCore/inspector/agents/page/PageDOMDebuggerAgent.h:
* Source/WebCore/inspector/agents/page/PageDOMDebuggerAgent.cpp:
(WebCore::PageDOMDebuggerAgent::disable):
(WebCore::PageDOMDebuggerAgent::mainFrameNavigated):
(WebCore::PageDOMDebuggerAgent::willFireAnimationFrame): Deleted.
(WebCore::PageDOMDebuggerAgent::didFireAnimationFrame): Deleted.
(WebCore::PageDOMDebuggerAgent::setAnimationFrameBreakpoint): Deleted.
* Source/WebCore/inspector/agents/worker/WorkerDOMDebuggerAgent.h:
* Source/WebCore/inspector/agents/worker/WorkerDOMDebuggerAgent.cpp:
(WebCore::WorkerDOMDebuggerAgent::setAnimationFrameBreakpoint): Deleted.
Move all `m_pauseOnAllAnimationFramesBreakpoint` code to the base class so it can be used inside `Worker`.

* Source/WebCore/inspector/agents/WebDebuggerAgent.h:
* Source/WebCore/inspector/agents/WebDebuggerAgent.cpp:
(WebCore::WebDebuggerAgent::didRequestAnimationFrame):
(WebCore::WebDebuggerAgent::willFireAnimationFrame):
(WebCore::WebDebuggerAgent::didCancelAnimationFrame):
(WebCore::WebDebuggerAgent::didFireAnimationFrame):
* Source/WebCore/inspector/agents/page/PageDebuggerAgent.h:
* Source/WebCore/inspector/agents/page/PageDebuggerAgent.cpp:
(WebCore::PageDebuggerAgent::didRequestAnimationFrame): Deleted.
(WebCore::PageDebuggerAgent::willFireAnimationFrame): Deleted.
(WebCore::PageDebuggerAgent::didCancelAnimationFrame): Deleted.
(WebCore::PageDebuggerAgent::didFireAnimationFrame): Deleted.
Move all animation frame async stack trace capture code to the base class so it can be used inside `Worker`.

* Source/WebInspectorUI/UserInterface/Controllers/DOMDebuggerManager.js:
(WI.DOMDebuggerManager.prototype._supportsAllAnimationFramesBreakpoint): Added.
(WI.DOMDebuggerManager.prototype._setEventBreakpoint):
(WI.DOMDebuggerManager.prototype._removeEventBreakpoint):
Allow the "All Animation Frames" breakpoint to be set on `Worker` targets.

* LayoutTests/inspector/worker/resources/worker-dom-debugger.js:
(functions.triggerAnimationFrame): Added.
* LayoutTests/inspector/worker/dom-debugger-event-animation-frame-breakpoints.html:
* LayoutTests/inspector/worker/dom-debugger-event-animation-frame-breakpoints-expected.txt:

* Source/JavaScriptCore/inspector/protocol/DOMDebugger.json:
* Source/WebInspectorUI/UserInterface/Protocol/Legacy/iOS/13.4/InspectorBackendCommands.js:
* Source/WebInspectorUI/UserInterface/Protocol/Legacy/iOS/14.0/InspectorBackendCommands.js:
* Source/WebInspectorUI/UserInterface/Protocol/Legacy/iOS/14.5/InspectorBackendCommands.js:
* Source/WebInspectorUI/UserInterface/Protocol/Legacy/iOS/15.0/InspectorBackendCommands.js:
* Source/WebInspectorUI/UserInterface/Protocol/Legacy/iOS/15.4/InspectorBackendCommands.js:
* Source/WebInspectorUI/UserInterface/Protocol/Legacy/iOS/16.0/InspectorBackendCommands.js:
* Source/WebInspectorUI/UserInterface/Protocol/Legacy/iOS/16.4/InspectorBackendCommands.js:
* Source/WebInspectorUI/UserInterface/Protocol/Legacy/iOS/17.0/InspectorBackendCommands.js:
* Source/WebInspectorUI/UserInterface/Protocol/Legacy/iOS/17.2/InspectorBackendCommands.js:
* Source/WebInspectorUI/UserInterface/Protocol/Legacy/iOS/17.4/InspectorBackendCommands.js:
* Source/WebInspectorUI/UserInterface/Protocol/Legacy/iOS/18.0/InspectorBackendCommands.js:
* Source/WebInspectorUI/UserInterface/Protocol/Legacy/iOS/18.2/InspectorBackendCommands.js:
* Source/WebInspectorUI/UserInterface/Protocol/Legacy/macOS/13.0/InspectorBackendCommands.js:
* Source/WebInspectorUI/UserInterface/Protocol/Legacy/macOS/13.3/InspectorBackendCommands.js:
* Source/WebInspectorUI/UserInterface/Protocol/Legacy/macOS/14.0/InspectorBackendCommands.js:
* Source/WebInspectorUI/UserInterface/Protocol/Legacy/macOS/14.2/InspectorBackendCommands.js:
* Source/WebInspectorUI/UserInterface/Protocol/Legacy/macOS/14.4/InspectorBackendCommands.js:
* Source/WebInspectorUI/UserInterface/Protocol/Legacy/macOS/15.0/InspectorBackendCommands.js:
* Source/WebInspectorUI/UserInterface/Protocol/Legacy/macOS/15.2/InspectorBackendCommands.js:
* Source/WebInspectorUI/Versions/Inspector-iOS-13.4.json:
* Source/WebInspectorUI/Versions/Inspector-iOS-14.0.json:
* Source/WebInspectorUI/Versions/Inspector-iOS-14.5.json:
* Source/WebInspectorUI/Versions/Inspector-iOS-15.0.json:
* Source/WebInspectorUI/Versions/Inspector-iOS-15.4.json:
* Source/WebInspectorUI/Versions/Inspector-iOS-16.0.json:
* Source/WebInspectorUI/Versions/Inspector-iOS-16.4.json:
* Source/WebInspectorUI/Versions/Inspector-iOS-17.0.json:
* Source/WebInspectorUI/Versions/Inspector-iOS-17.2.json:
* Source/WebInspectorUI/Versions/Inspector-iOS-17.4.json:
* Source/WebInspectorUI/Versions/Inspector-iOS-18.0.json:
* Source/WebInspectorUI/Versions/Inspector-iOS-18.2.json:
* Source/WebInspectorUI/Versions/Inspector-macOS-13.0.json:
* Source/WebInspectorUI/Versions/Inspector-macOS-13.3.json:
* Source/WebInspectorUI/Versions/Inspector-macOS-14.0.json:
* Source/WebInspectorUI/Versions/Inspector-macOS-14.2.json:
* Source/WebInspectorUI/Versions/Inspector-macOS-14.4.json:
* Source/WebInspectorUI/Versions/Inspector-macOS-15.0.json:
* Source/WebInspectorUI/Versions/Inspector-macOS-15.2.json:
* LayoutTests/inspector/worker/dom-debugger-dom-breakpoints.html: Removed.
* LayoutTests/inspector/worker/dom-debugger-dom-breakpoints-expected.txt: Removed.
* LayoutTests/platform/mac-site-isolation/TestExpectations:

Drive-by: mark `DOMDebugger.setDOMBreakpoint` and `DOMDebugger.removeDOMBreakpoint` as being only supported by `"page"` targets since that's how the code behaves (i.e. it'll throw an error if dispatched to a `Worker`).
Canonical link: https://commits.webkit.org/290245@main
